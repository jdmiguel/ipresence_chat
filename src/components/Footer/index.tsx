import React from 'react';
interface IFooterProps {
  onChange: (event: React.ChangeEvent) => void;
  onClick: () => void;
};

const Footer:React.FC<IFooterProps> = ({onChange, onClick}) => {
  return(
    <footer>
      <div className="footer-wrapper">
        <textarea placeholder="Send a message..." onChange={onChange}/>
        <button type="button" onClick={onClick}>
          <i className="material-icons">
            send
          </i>
        </button>
      </div>
    </footer>
  )
}

export default Footer;