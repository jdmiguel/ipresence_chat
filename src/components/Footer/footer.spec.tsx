import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Footer from '.';

const FOOTER_PROPS = {
  currentMessage: '',
  onClickButton: () => {},
  onChangeMessage: () => {}
};

describe('Component: Footer', () => {
  it('should render and display deactive button', () => {
    const { container, getByTestId } = render(
      <Footer {...FOOTER_PROPS} />
    );

    expect(getByTestId('footer-button')).not.toHaveClass('active');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onClickButton function', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Footer {...FOOTER_PROPS} onClickButton={handleClick}/>
    );
    const button = getByTestId('footer-button');

    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalled();
  });

  it('should call onChangeMessage function', () => {
    const handleChange = jest.fn();
    const { getByPlaceholderText } = render(
      <Footer {...FOOTER_PROPS} onChangeMessage={handleChange}/>
    );

    fireEvent.change(getByPlaceholderText('Send a message...'), { target: { value: 'hi' } });
    expect(handleChange).toHaveBeenCalled();
  });

  it('should button be active', async () => {
    const { getByTestId } = render(
      <Footer {...FOOTER_PROPS} currentMessage="hi"/>
    );

    expect(getByTestId('footer-button')).toHaveClass('active');
  });
});