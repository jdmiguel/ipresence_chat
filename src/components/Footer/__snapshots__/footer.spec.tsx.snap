// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer should render and display deactive button 1`] = `
<footer>
  <div
    class="footer-wrapper"
  >
    <textarea
      placeholder="Send a message..."
    />
    <button
      class=""
      data-testid="footer-button"
      type="button"
    >
      <i
        class="material-icons"
      >
        send
      </i>
    </button>
  </div>
</footer>
`;
